medusa/storage/azure_blobs_storage/azcli.py

```diff
@@ -0,0 +1,126 @@
+import io
+import json
+import logging
+import os
+import subprocess
+import sys
+import uuid
+
+from retrying import retry
+
+
+class AzCli(object):
+    def __init__(self, storage):
+        self._config = storage.config
+        self.storage = storage
+
+    @property
+    def bucket_name(self):
+        return self._config.bucket_name
+
+    def __enter__(self):
+        with io.open(os.path.expanduser(self._config.key_file), 'r', encoding='utf-8') as json_fi:
+            credentials = json.load(json_fi)
+        self._env = dict(
+            os.environ,
+            AZURE_STORAGE_ACCOUNT=credentials['storage_account'],
+            AZURE_STORAGE_KEY=credentials['key']
+        )
+        self._az_cli_path = self.find_az_cli()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._env = dict(os.environ)
+        return False
+
+    @staticmethod
+    def find_az_cli():
+        """
+        Construct the AZ command line with parameters and variables
+        Also includes a lookup for the AZ binary, in case we are running
+        under a venv
+        """
+        az_bin = 'az'
+        binary_paths = ['/usr/bin', '/usr/local/bin']
+        paths = sys.path + binary_paths
+        for path in paths:
+            if not path:
+                continue
+            tpath = '/'.join([path, 'az'])
+            if os.path.exists(tpath) and os.path.isfile(tpath) and os.access(tpath, os.X_OK):
+                az_bin = tpath
+                break
+        return az_bin
+
+    def cp_upload(self, *, srcs, bucket_name, dest, max_retries=5):
+        job_id = str(uuid.uuid4())
+        azcli_output = "/tmp/azcli_{0}.output".format(job_id)
+        objects = []
+        for src in srcs:
+            cmd = [self._az_cli_path, "storage", "blob", "upload", "-f", str(src), "-c", bucket_name, "-n", dest]
+            objects.append(self.upload_file(cmd, dest, azcli_output))
+
+        return objects
+
+    def cp_download(self, *, src, bucket_name, dest, max_retries=5):
+        job_id = str(uuid.uuid4())
+        azcli_output = "/tmp/azcli_{0}.output".format(job_id)
+        objects = []
+        cmd = [self._az_cli_path, "storage", "blob", "download", "-f", str(dest), "-c", bucket_name, "-n", src]
```

Comment:
```suggestion
        cmd = [self._az_cli_path, "storage", "blob", "download", "-f", os.path.join(str(dest), str(src).split("/")[-1]), "-c", bucket_name, "-n", src]
```

The typo fix from `downlod_blobs` to `download_blobs` along with making multi part upload threshold configurable made this code path testable during integration tests.
It fails consistently with the following error:
```
CLIInternalError: The command failed with an unexpected error. Here is the traceback:
[Errno 21] Is a directory: '/tmp/medusa-download-all-tables/system_auth/roles-5bc52802de2535edaeab188eecebb090/'
Traceback (most recent call last):
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 654, in execute
    raise ex
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 709, in _run_job
    cmd_copy.exception_handler(ex)
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/command_modules/storage/__init__.py", line 341, in new_handler
    raise ex
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 688, in _run_job
    result = cmd_copy(params)
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 325, in __call__
    return self.handler(*args, **kwargs)
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 784, in default_command_handler
    return op(**command_args)
  File "/Users/adejanovski/projets/cassandra/thelastpickle/py36/lib/python3.6/site-packages/azure/multiapi/storage/v2018_11_09/blob/baseblobservice.py", line 1968, in get_blob_to_path
    with open(file_path, open_mode) as stream:
IsADirectoryError: [Errno 21] Is a directory: '/tmp/medusa-download-all-tables/system_auth/roles-5bc52802de2535edaeab188eecebb090/'
To open an issue, please run: 'az feedback'
```

Apparently the destination path must contain the file name and not just the destination directory name.
With the above suggestion, integration tests are passing.